gensim/scripts/segment_wiki.py

```diff
@@ -141,16 +146,18 @@ def extract_page_xmls(f):
         XML strings for page tags.
 
     """
-    elems = (elem for _, elem in cElementTree.iterparse(f, events=("end",)))
+    elems = cElementTree.iterparse(f, events=("end",))
 
-    elem = next(elems)
+    _, elem = next(elems)
     namespace = get_namespace(elem.tag)
     ns_mapping = {"ns": namespace}
     page_tag = "{%(ns)s}page" % ns_mapping
 
-    for elem in elems:
+    for _, elem in elems:
         if elem.tag == page_tag:
-            yield cElementTree.tostring(elem)
+            # Pickle object explicitly to send it to the children
+            # as elem.clear() will be called earlier than implicitly serialization of elem
+            yield pickle.dumps(elem)
```

Comment:
Does using a higher pickle protocol (not the default ASCII) make sense?

Reply:
I tried but don't find some meaningful profit.